runtime/interpreter/encode.go

```diff
@@ -1601,6 +1622,21 @@ var _ atree.TypeInfo = compositeTypeInfo{}
 
 const encodedCompositeTypeInfoLength = 3
 
+func (c compositeTypeInfo) IsComposite() bool {
+	return true
+}
+
+func (c compositeTypeInfo) Identifier() string {
+	// TODO: maybe improve this to be more efficient, etc.
+	return "composite(" + c.location.ID() + "/" + c.qualifiedIdentifier + ")"
```

Comment:
Can this maybe just be `c.Location.TypeID(gauge, c.qualifiedIdentifier)`?

Reply:
It's not ideal. Theoretically we need to meter all allocations. We've seen this issue of atree calling out to Cadence without providing e.g. access to a necessary memory gauge or interpreter a few times before, so I think we should maybe discuss how we can to address this in general (probably as a follow-up and not part of this PR)

Reply:
If we can do that as a followup let's please do that.

Reply:
Improved this in da81afe818f6f126d3b37e11b003bf8af8db9626, using the un-metered approach for now